src/lucky_router/fragment.cr

```diff
@@ -59,12 +59,24 @@ class LuckyRouter::Fragment(T)
   # the final step to finding the payload is to search for a matching request method
   getter method_to_payload = Hash(String, T).new
 
+  def find(parts : Array(String), method : String) : Match(T) | NoMatch
+    MatchFinder(T).new(self, parts: parts, method: method).run
+  end
+
   def process_parts(parts : Array(String), method : String, payload : T)
-    PartProcessor(T).new(self, parts: parts, method: method, payload: payload).run
-    self
+    leaf_fragment = parts.reduce(self) { |fragment, part| fragment.add_part(part) }
+    leaf_fragment.method_to_payload[method] = payload
   end
 
-  def find(parts : Array(String), method : String) : Match(T) | NoMatch
-    MatchFinder(T).new(self, parts: parts, method: method).run
+  def add_part(part : String) : Fragment(T)
+    if part.starts_with?(":")
+      dynamic_fragment = self.dynamic_part ||= DynamicFragment(T).new(
+        name: part[1...],
+        fragment: Fragment(T).new
+      )
```

Comment:
It took me awhile to figure out what what this was doing and why and then realized it was removing the first character which is the `:`. Thoughts on adding this explaining variable to clarify what it is doing?

```suggestion
      part_without_colon = part[1..]
      dynamic_fragment = self.dynamic_part ||= DynamicFragment(T).new(
        name: part_without_colon,
        fragment: Fragment(T).new
      )
```

Reply:
This kind of thing is why I want to wrap the part strings in a part class.

Reply:
Yeah I was thinking that when I read your comment. That will be a really awesome refactor 👍 Nice idea